src/endpoints/ui.rs

```diff
@@ -169,25 +210,16 @@ async fn parse_form<T: for<'de> Deserialize<'de>>(
     let mut qs = String::new();
     while let Some(field) = form.next_field().await? {
         let name = field.name().ok_or(ParseFormError::NameMissing)?;
-        let name = urlencoding::encode(name);
+        let name = urlencoding::encode(name).to_string();
         match field.file_name() {
             Some(filename) if !filename.is_empty() => {
-                let id = Uuid::new_v4();
-                if filename.contains('/') {
-                    return Err(ParseFormError::FilenameSlash(filename.to_string()));
-                }
-                let filename_escaped = urlencoding::encode(filename);
+                let file = stream_field_to_file(field, files_dir).await?;
+                let filename_escaped = urlencoding::encode(&file.name);
+                let id = file.id;
```

Comment:
Have you tested that `/` is escaped? Or is this verified elsewhere?

Reply:
This is now done within `stream_field_to_file`, I think.